pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -224,3 +245,43 @@ func checkValidPatchTypes(patches []hivev1.SyncObjectPatch) (string, bool) {
 	}
 	return "", true
 }
+
+func checkSecretReferences(secrets []hivev1.SecretReference) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, secret := range secrets {
+		sourcePath := field.NewPath("spec").Child("secretReferences").Index(i).Child("source")
+		if len(secret.Source.Kind) == 0 {
+			allErrs = append(allErrs, field.Required(sourcePath.Child("kind"), "Kind is required"))
+		}
+		if secret.Source.Kind != "secret" {
+			allErrs = append(allErrs, field.NotSupported(sourcePath.Child("kind"), secret.Source.Kind, []string{"secret"}))
+		}
+		if len(secret.Source.Name) == 0 {
+			allErrs = append(allErrs, field.Required(sourcePath.Child("name"), "Name is required"))
+		}
+		if len(secret.Source.FieldPath) != 0 {
+			allErrs = append(allErrs, field.Invalid(sourcePath.Child("fieldPath"), secret.Source.FieldPath, "FieldPath should not be set"))
+		}
+		if len(secret.Source.UID) != 0 {
+			allErrs = append(allErrs, field.Invalid(sourcePath.Child("UID"), secret.Source.UID, "UID should not be set"))
+		}
+
+		targetPath := field.NewPath("spec").Child("secretReferences").Index(i).Child("target")
+		if len(secret.Target.Kind) == 0 {
```

Comment:
How do we handle the API version of the target? Is there any concern that the target may not be able to accept the version sent from the controller?

Reply:
Based on Friday convo, I believe we're deferring worrying about the `APIVersion` of the source not matching the target.